pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -436,10 +624,31 @@ func generatePeerConfigs(subnets []string, gatewayIP net.IP) (map[*net.IPNet]net
 	return peerConfigs, nil
 }
 
-// getPeerGatewayIP will always return the first Gateway IP.
-func getPeerGatewayIP(spec mcv1alpha1.ClusterInfo) net.IP {
+// getPeerGatewayIP will return the network IP address of the Service CIDR when WireGuard enabled,
```

Comment:
I did not understand this? Could you explain?

And probably we should change a name to better indicate the func's function.

Reply:
I think we should rename the func to `getPeerGatewayTunnelIP`, and please add add some comments to explain this. I still do not fully get why the first IP in ServiceCIDR is used. Please try to explain in your comments.

And could you also help rename `l3FwdFlowToRemoteViaTun` in `openflow/multicluster.go` to `l3FwdFlowToRemoteGateway`? The current name is misleading given the same name is used for local cluster tunnel flows.

Reply:
Renamed the two functions and revised the comment to explain why we need to calculate the first IP of the ServiceCIDR.